client/gui/CAnimation.h

```diff
@@ -9,16 +9,16 @@
  */
 #pragma once
 
+#include "../../lib/GameConstants.h"
 #include "../../lib/vcmi_endian.h"
 #include "Geometries.h"
-#include "../../lib/GameConstants.h"

```

Comment:
is it also something format-related?

Reply:
What exactly? Header reordering?
Yes, clang-format can rearrange includes. Not even sure whether I can disable it completely, but current rules are:
- StdInc.h always comes first
- Main include (include of header with same name as cpp name) always comes after StdInc.h)
- After this, clang-format will detect "groups" - continuous includes without empty lines and sort them alphabetically
- All group will be preserved - clang will not try to rearrange them

Reply:
"And sort them alphabetically" - is it possible to remove this part?
Actually it's bug-prone, as you may include header which has include inside.

Reply:
Ah, it looks that I can I think.

And how it is bug-prone? The only exception is StdInc.h since it includes commonly used headers.
Other than that - order of headers does not matters (assuming well-written code)

Reply:
You may have implicit declaration of class in header, included into other header. It means, that you won't have comprehensive list of headers.

Reply:
Then this is error in code. Header must have complete set of all required forward declarations.

Reply:
This error is hidden, because you will not have any warning or compilation error. "Classic" example
```
#include <vector>
#include "header.h" //vector is not included but used
```
vs
```
#include "header.h" //compilation error
#include <vector>
```

Reply:
Again, this is an error in header.h. 
header must include all its dependencies. If not - this is a bug.

Reply:
I like sorted includes

> header must include (or forward-declare if applicable) all its dependencies. If not - this is a bug.

true